google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/DefaultCloudSdkService.java

```diff
@@ -40,4 +79,168 @@ public String getCloudSdkHomePath() {
   public void setCloudSdkHomePath(String cloudSdkHomePath) {
     propertiesComponent.setValue(CLOUD_SDK_PROPERTY_KEY, cloudSdkHomePath);
   }
+
+  @NotNull
+  @Override
+  public File getToolsApiJarFile() {
+    return new File(JAVA_TOOLS_BASE_PATH,
+        JpsAppEngineModuleExtensionImpl.LIB_APPENGINE_TOOLS_API_JAR);
+  }
+
+  @NotNull
+  @Override
+  public File[] getLibraries() {
+    return getJarsFromDirectory(new File(JAVA_TOOLS_BASE_PATH, "lib/shared"));
+  }
+
+  @NotNull
+  @Override
+  public File[] getJspLibraries() {
+    return getJarsFromDirectory(new File(JAVA_TOOLS_BASE_PATH, "lib/shared/jsp"));
+  }
+
+  @Override
+  public boolean isMethodInBlacklist(@NotNull String className, @NotNull String methodName) {
+    if (myMethodsBlackList == null) {
+      try {
+        myMethodsBlackList = loadBlackList();
+      } catch (IOException ioe) {
+        logger.error(ioe);
+        myMethodsBlackList = new THashMap<>();
+      }
+    }
+    final Set<String> methods = myMethodsBlackList.get(className);
+    return methods != null && methods.contains(methodName);
+  }
+
+  @Override
+  public boolean isClassInWhiteList(@NotNull String className) {
+    return AppEngineJreWhitelist.contains(className);
+  }
+
+  @NotNull
+  @Override
+  public List<String> getUserLibraryPaths() {
```

Comment:
Since the libraries are under the Cloud SDK folder, in this case, we could add "appengine-endpoints" and "jsr107" to the keys returned by getJarPath(). We wouldn't need to assume that those jars are located at JAVA_TOOLS_BASE_PATH + "lib/opt/user", for example.